pkg/function-manager/handlers.go

```diff
@@ -121,9 +122,14 @@ func functionModelOntoEntity(m *v1.Function, e *functions.Function) error {
 	if m.Main != nil && *m.Main != "" {
 		main = *m.Main
 	}
+	if m.Image != nil && *m.Image != "" {
+		e.ImageName = *m.Image
+	}
+	if m.Runtime != nil && *m.Runtime != "" {
```

Comment:
We will likely want validation somewhere that fails if Image and Runtime are empty.

Reply:
added in `controller.go` in the `Add` method